src/main/java/com/es/phoneshop/web/ProductDetailsPageServlet.java

```diff
@@ -2,30 +2,91 @@
 
 import com.es.phoneshop.model.dao.impl.ArrayListProductDao;
 import com.es.phoneshop.model.dao.ProductDao;
-import com.es.phoneshop.model.entity.sortParams.SortField;
-import com.es.phoneshop.model.entity.sortParams.SortOrder;
+import com.es.phoneshop.model.entity.cart.Cart;
+import com.es.phoneshop.model.entity.cart.CartService;
+import com.es.phoneshop.model.entity.cart.DefaultCartService;
+
+import com.es.phoneshop.model.entity.latestProductQueue.DefaultLatestProductQueueService;
+import com.es.phoneshop.model.entity.latestProductQueue.LatestProductQueue;
+import com.es.phoneshop.model.entity.latestProductQueue.LatestProductQueueService;
+import com.es.phoneshop.model.exceptions.OutOfStockException;
 
 import javax.servlet.ServletConfig;
 import javax.servlet.ServletException;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.io.IOException;
-import java.util.Optional;
+import java.text.NumberFormat;
+import java.text.ParseException;
 
 
 public class ProductDetailsPageServlet extends HttpServlet {
     private ProductDao productDao;
+    private CartService cartService;
+    private LatestProductQueueService queueService;
 
     @Override
     public void init(ServletConfig config) throws ServletException {
         super.init(config);
         productDao = ArrayListProductDao.getInstance();
+        cartService = DefaultCartService.getInstance();
+        queueService = DefaultLatestProductQueueService.getInstance();
     }
     @Override
     protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
-        String productId = request.getPathInfo();
-        request.setAttribute("product", productDao.getProduct(Long.valueOf(productId.substring(1))));
+        request.setAttribute("product", productDao.getProduct(parseProductId(request)));
+        request.setAttribute("cart", cartService.getCart(request));
+        request.setAttribute("latestProducts", queueService.getLatestProductQueue(request).getQueue());
+
+        LatestProductQueue queue = queueService.getLatestProductQueue(request);
+        queueService.add(queue, parseProductId(request));
         request.getRequestDispatcher("/WEB-INF/pages/product.jsp").forward(request, response);
     }
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        String quantity = request.getParameter("quantity");
+        Long id = parseProductId(request);
+        int parsedQuantity = 0;
+        try {
+            NumberFormat format = NumberFormat.getInstance(request.getLocale());
```

Comment:
![image](https://user-images.githubusercontent.com/16850376/216260373-b05c713c-80b5-44c9-aa0a-60689514a909.png)
Highlited quantity can be added to cart